python/vowpalwabbit/DFtoVWtoTensorboard.py

```diff
@@ -0,0 +1,182 @@
+		print('label:', label, end='\t')    
+		print('prediction:', prediction, end='\t')    
+		print('num_features:', num_features) 
+
+
+	#------------------------------------------------------------------------------------------------		
+	def fit(self, vw_to_tensorboard=None):
+		"""Learns on the relevant examples and can also log metrics for tensorboard visualization
+
+		Parameters
+		----------
+
+		vw_to_tensorboard  : VWtoTensorboard object 
+								Default value is None, this parameter is used to control the logging of metrics for Tensorboard visualization
+								If value is VWtoTensorboard object : metrics are logged for Tensorboard visualization
+								If value is None : metrics are not logged for Tensorboard visualization
+
+		Returns
+		-------
+
+		None
+		"""
+		if isinstance(vw_to_tensorboard, VWtoTensorboard):
+			file_writer = tx.SummaryWriter(vw_to_tensorboard.logdir)   # creating file writer
+		
+		sum_loss = 0.
+		weighted_examples = 0.
+
+		for iteration, vw_format in enumerate(self.vw_formatted_data):
+			example = self.vw.parse(vw_format)       # parse the string format, it returns an example object
+			self.vw.learn(example)                  # learn on example
+
+			label = pyvw.get_label(example, self.vw.get_label_type())
+			prediction = pyvw.get_prediction(example, self.vw.get_prediction_type())
+			num_features = example.get_feature_number()    
+
+			self.vw.finish_example(example)  # Any use of vw object should be done after this and use of example before this 
+
+			sum_loss_since_last = self.vw.get_sum_loss() - sum_loss  # vw.get_sum_loss() return current sum loss, sum_loss variable right now holds sum loss of previous iteration
+			weighted_examples_since_last = self.vw.get_weighted_examples() - weighted_examples  # vw.get_weighted_examples() return current weighted examples(sum),  weighted_examples variable right now holds weighted examples of previous iteration		        
+		
+			sum_loss = self.vw.get_sum_loss()  # Now sum_loss no longer hold previous iteration's sum_loss
+			weighted_examples = self.vw.get_weighted_examples()  # Now weighted_examples no longer hold previous iteration's weighted examples
+		
+			average_loss = self._calculate_average_loss(sum_loss, weighted_examples)
+			since_last = self._calculate_since_last(sum_loss_since_last, weighted_examples_since_last)
+		
+			self._print_metrics(average_loss, since_last, label, prediction, num_features)
+
+			if isinstance(vw_to_tensorboard, VWtoTensorboard):
+				file_writer.add_scalar('average_loss', average_loss, iteration)  # logging average_loss on each iteration
+				file_writer.add_scalar('since_last', since_last, iteration)   # logging since_last on each iteration
+			#     file_writer.add_scalar('label' , label, iteration)
+			#     file_writer.add_scalar('prediction', prediction, iteration)
+			#     file_writer.add_histogram('label-prediction', [label, prediction], iteration)
+			#     file_writer.add_histogram('label', label, iteration)
```

Comment:
[Minor] When you are ready to get this merged, remember to remove commented out code.